hawkbit-rest/hawkbit-ddi-resource/src/test/java/org/eclipse/hawkbit/ddi/rest/resource/DdiConfirmationBaseTest.java

```diff
@@ -0,0 +1,272 @@
+
+    @Test
+    @Description("Forced deployment to a controller. Checks if the confirmation resource response payload for a given" +
+            " deployment is as expected.")
+    public void deploymentForceAction() throws Exception {
+        enableUserConsentFlow();
+        // Prepare test data
+        final DistributionSet ds = testdataFactory.createDistributionSet("", true);
+        final DistributionSet ds2 = testdataFactory.createDistributionSet("2", true);
+        final Artifact artifact = testdataFactory.createArtifact(RandomUtils.nextBytes(ARTIFACT_SIZE), getOsModule(ds),
+                "test1", ARTIFACT_SIZE);
+        final Artifact artifactSignature = testdataFactory.createArtifact(RandomUtils.nextBytes(ARTIFACT_SIZE),
+                getOsModule(ds), "test1.signature", ARTIFACT_SIZE);
+
+        final Target savedTarget = testdataFactory.createTarget(DdiConfirmationBaseTest.DEFAULT_CONTROLLER_ID);
+        assertThat(deploymentManagement.findActiveActionsByTarget(PAGE, savedTarget.getControllerId())).isEmpty();
+
+
+        final List<Target> targetsAssignedToDs = assignDistributionSet(ds.getId(), savedTarget.getControllerId(),
+                Action.ActionType.FORCED).getAssignedEntity().stream().map(Action::getTarget).collect(Collectors.toList());
+
+        assertThat(deploymentManagement.findActiveActionsByTarget(PAGE, savedTarget.getControllerId())).hasSize(1);
+
+        final Action action = deploymentManagement.findActiveActionsByTarget(PAGE, savedTarget.getControllerId())
+                .getContent().get(0);
+        assertThat(deploymentManagement.countActionsAll()).isEqualTo(1);
+
+        assignDistributionSet(ds2, targetsAssignedToDs).getAssignedEntity();
+        assertThat(deploymentManagement.findActiveActionsByTarget(PAGE, savedTarget.getControllerId())).hasSize(2);
+        assertThat(deploymentManagement.countActionsAll()).isEqualTo(2);
+
+        final Action uaction = deploymentManagement.findActiveActionsByTarget(PAGE, savedTarget.getControllerId())
+                .getContent().get(0);
+        assertThat(uaction.getDistributionSet()).isEqualTo(ds);
+        assertThat(deploymentManagement.findActiveActionsByTarget(PAGE, savedTarget.getControllerId())).hasSize(2);
+
+        // Run test
+        final long current = System.currentTimeMillis();
+        final String expectedConfirmationBaseLink = String.format("/%s/controller/v1/%s/confirmationBase/%d",
+                tenantAware.getCurrentTenant(), DEFAULT_CONTROLLER_ID, uaction.getId());
+
+        performGet(CONTROLLER_BASE, MediaTypes.HAL_JSON, status().isOk(), tenantAware.getCurrentTenant(),
+                DEFAULT_CONTROLLER_ID).andExpect(jsonPath("$.config.polling.sleep", equalTo("00:01:00")))
+                .andExpect(jsonPath("$._links.confirmationBase.href", containsString(expectedConfirmationBaseLink)))
+                .andExpect(jsonPath("$._links.deploymentBase.href").doesNotExist());
```

Comment:
Please verify in an additional step, that the deploymentBase endpoint is not accessible for this action. Add one more test, where the action is in RUNNING state and the confirmationBase is not accessible.

Reply:
Done.

Reply:
Implemented a separate test case to test that confirmation endpoint is not available when user content flow feature is not toggled on.